app/[lang]/chapters/[slug]/[lesson]/page.tsx

```diff
@@ -66,65 +67,45 @@ export default function Page({ params }) {
   }
 
   // If the lesson does not exist, we show this error message.
-  if (!(params.lesson in chapterLessons) || !(params.slug in chapters)) {
-    return (
-      <>
-        <Head />
-        <div className="flex h-full w-full grow flex-col items-center justify-center">
-          <span className="mb-10 font-nunito text-2xl text-white">
-            Sorry, we could not find the ’{params.lesson}’ lesson.
-          </span>
-          <Button
-            href={`/${params.lang}/chapters/${params.slug}/intro-1`}
-            size="small"
-          >
-            &larr; Back to Chapter Overview
-          </Button>
-        </div>
-      </>
-    )
+  if (
+    (chapterLessons && !(params.lesson in chapterLessons)) ||
+    !(params.slug in chapters)
+  ) {
+    return notFound()
   }
 
   const Lesson = chapterLessons[params.lesson].default
 
   if (dev) {
     return (
-      <>
-        <Head />
-        <Lesson lang={params.lang} />
-      </>
+      hydrated && (
```

Comment:
Is there any particular reason we want to wait for this component to be hydrated, seems like all the data that is being rendered should be available on intial render?